aries_vcx/src/handlers/issuance/credential_def/revocation_registry.rs

```diff
@@ -0,0 +1,59 @@
+use crate::error::{VcxErrorKind, VcxResult};
+use crate::handlers::issuance::credential_def::PublicEntityStateType;
+use crate::libindy::utils::anoncreds;
+use crate::libindy::utils::anoncreds::RevocationRegistryDefinition;
+
+#[derive(Clone, Deserialize, Debug, Serialize, PartialEq)]
+pub struct RevocationRegistry {
+    pub rev_reg_id: String,
+    pub(super) rev_reg_def: RevocationRegistryDefinition,
+    pub(super) rev_reg_entry: String,
+    pub(super) tails_base_dir: String,
+    pub(super) max_creds: u32,
+    pub(super) tag: u32,
+    pub(super) state_rev_reg_def: PublicEntityStateType,
+    pub(super) state_rev_reg_delta: PublicEntityStateType,
+}
+
+
+impl RevocationRegistry {
+    pub fn create(issuer_did: &str, cred_def_id: &str, tails_base_dir: &str, max_creds: u32, tag_no: u32) -> VcxResult<RevocationRegistry> {
+        let (rev_reg_id, rev_reg_def, rev_reg_entry) =
+            anoncreds::generate_rev_reg(&issuer_did, &cred_def_id, tails_base_dir, max_creds, &format!("tag{}", tag_no))
+                .map_err(|err| err.map(VcxErrorKind::CreateRevRegDef, "Cannot create Revocation Registry"))?;
+        Ok(RevocationRegistry {
+            rev_reg_id,
+            rev_reg_def,
+            rev_reg_entry,
+            tails_base_dir: tails_base_dir.to_string(),
+            max_creds,
+            tag: tag_no,
+            state_rev_reg_def: PublicEntityStateType::Built,
+            state_rev_reg_delta: PublicEntityStateType::Built,
+        })
+    }
+
+    pub fn publish_rev_reg_def(mut self, issuer_did: &str, tails_url: &str) -> VcxResult<Self> {
+        trace!("publish_rev_reg_def >>> Publish revocation registry. issuer_did:{}, rev_reg_id: {}, rev_reg_def:{:?}", issuer_did, &self.rev_reg_id, &self.rev_reg_def);
+        self.rev_reg_def.value.tails_location = String::from(tails_url);
+        anoncreds::publish_rev_reg_def(&issuer_did, &self.rev_reg_def)
+            .map_err(|err| err.map(VcxErrorKind::InvalidState, "Cannot publish revocation registry definition"))?;
+        let mut new_rev_reg = Self {
+            state_rev_reg_def: PublicEntityStateType::Published,
+            ..self
+        };
+        Ok(new_rev_reg)
+    }
+
+    pub fn publish_rev_reg_delta(self, issuer_did: &str) -> VcxResult<Self> {
+        trace!("publish_rev_reg_delta >>> Publishing initial revocation registry delta. issuer_did:{}, rev_reg_id: {}", issuer_did, self.rev_reg_id);
```

Comment:
Perhaps `RevocationRegistry::publish_rev_reg_delta >>> issuer_did: {}, rev_reg_id: {}` to be consistent